lib/services/schema-object-factory.ts

```diff
@@ -294,7 +294,7 @@ export class SchemaObjectFactory {
 
     if (!(enumName in schemas)) {
       schemas[enumName] = {
-        type: 'string',
+        type: metadata.type as string,
```

Comment:
```suggestion
        type: metadata.type as string ?? 'string',
```
Hmm?

Reply:
Currently, the `metadata.type` cannot take on the values of `undefined` or `null`:
https://github.com/nestjs/swagger/blob/cf4b40eb2640ce42283cdaf1644145d0cf6a22e2/lib/utils/enum.utils.ts#L36-L39

Nonetheless, I do agree that adding a safeguard would be a good idea to prepare for any future changes.